src/common/errors.ts

```diff
@@ -51,6 +51,45 @@ ${raw ? `Response: ${JSON.stringify(raw, null, 2)}` : ""}
   }
 }
 
+type ErrorBody = {
+  code?: string;
+  name?: string;
+  severity?: string;
+  source?: string;
+  description?: string;
+  resolution?: string;
+  helpLink?: string;
+  location?: {
+    failingLine?: number;
+    startOffset?: number;
+    endOffset?: number;
+  };
+};
+
+export class CompositeError extends Error {
+  message: string;
+  errorsArray: Array<ErrorBody>;
+
+  constructor(errors: Array<ErrorBody>) {
+    const parsedErrors = errors.map(error => {
+      const severity = error.severity ? `${error.severity}: ` : "";
```

Comment:
could you add more tests that validate all these ternary operatoers? This will significantly improve the test coverage.

Reply:
We were at high-80%/low-90%, but sure. Added more tests to make it 98%, hopefully that's good enough.